### PR TITLE
Creating package dependency without a version will result in "latest"

### LINKED ISSUE #75: PackageContext.Install(PackageDependency) with version 'null' should return the latest version
This method used to work with "latest", but now doesn't:
`PackageContext.Install(PackageDependency)`
When fed with `PackageDependency(name: hl7.fhir.r4.core, version: null)`
It now returns
`Package 'hl7.fhir.r4.core (latest)' was not found.`

A work-around exists, to manually change null to "Latest". But if you look at type (PackageDependency), null must be interpreted as latest by the library. So there might be other regressions elsewhere.

### PR DESCRIPTION
**Pull request by mmsmits**
fixes #75

### CODE REVIEW

File: Firely.Fhir.Packages/Source/FhirPackageSource.cs [hunk 1]
```diff
@@ -90,14 +90,7 @@ static IEnumerable<PackageDependency> createDependencies(string[] dependencies)
...
                 {
-                    var splitDependency = dep.Split('@');
-                    if (splitDependency.Length == 1)
-                        yield return new PackageDependency(dep, "latest");
-                    else
-                    {
-                        var versionDep = splitDependency[1];
-                        yield return new PackageDependency(splitDependency[0], versionDep);
-                    }
+                    yield return (PackageDependency)dep;
```
Comment (ewoutkramer) [inline Firely.Fhir.Packages/Source/FhirPackageSource.cs:93]: Isn't this just `dependencies.Cast<PackageDependency>()` ?

**Review by ewoutkramer — CHANGES_REQUESTED** [review]

File: Firely.Fhir.Packages/Source/FhirPackageSource.cs (hunk 1 above)
Comment (mmsmits) [reply Firely.Fhir.Packages/Source/FhirPackageSource.cs:93]: That does not work for user-defined operators sadly:
https://stackoverflow.com/questions/13316718/explicit-implicit-cast-operator-fails-when-using-linqs-cast-operator

Comment (ewoutkramer) [reply Firely.Fhir.Packages/Source/FhirPackageSource.cs:93]: I don't like this cast operator magic anyway, it's basically hiding the creation of a PackageDependency object here.....why not just  `new PackageDependency(dep)` ? Everyone recognizes that....

Comment (ewoutkramer) [reply Firely.Fhir.Packages/Source/FhirPackageSource.cs:93]: But ok, that's a bigger change.

**Review by ewoutkramer — APPROVED** [review]

**Pull request merged**